llvm/lib/Target/AArch64/Disassembler/AArch64Disassembler.cpp

```diff
@@ -488,6 +491,20 @@ static DecodeStatus DecodePCRelLabel19(MCInst &Inst, unsigned Imm,
   return Success;
 }
 
+static DecodeStatus DecodePCRelLabel9(MCInst &Inst, unsigned Imm, uint64_t Addr,
+                                      const MCDisassembler *Decoder) {
+  int64_t ImmVal = Imm;
+
+  // Sign-extend 9-bit immediate.
+  if (ImmVal & (1 << (9 - 1)))
+    ImmVal |= ~((1LL << 9) - 1);
+
+  if (!Decoder->tryAddingSymbolicOperand(Inst, (ImmVal * 4), Addr,
```

Comment:
LGTM.

It'd be very nice to include in the commit message information about what was fixed, e.g.
"Replaced a hift left of a possibly negative value with a multiplication in DecodePCRelLabel9"